src/periodograms.jl

```diff
@@ -73,53 +72,51 @@ function fft2pow!(out::AbstractArray{T}, s_fft::AbstractVector{Complex{T}}, nfft
     n = length(s_fft)
     if onesided
         m2 = convert(T, 2/r)
-        out[offset+1] += abs2(s_fft[1])*m1
+        out[offset+1] = muladd(abs2(s_fft[1]), m1, out[offset+1])
         for i = 2:n-1
-            @inbounds out[offset+i] += abs2(s_fft[i])*m2
+            @inbounds out[offset+i] = muladd(abs2(s_fft[i]), m2, out[offset+i])
         end
-        out[offset+n] += abs2(s_fft[end])*ifelse(iseven(nfft), m1, m2)
+        out[offset+n] = muladd(abs2(s_fft[end]), ifelse(iseven(nfft), m1, m2), out[offset+n])
     else
         if n == nfft
-            for i = 1:length(s_fft)
-                @inbounds out[offset+i] += abs2(s_fft[i])*m1
+            for i in eachindex(s_fft)
+                @inbounds out[offset+i] = muladd(abs2(s_fft[i]), m1, out[offset+i])
             end
         else
             # Convert real FFT to two-sided
-            out[offset+1] += abs2(s_fft[1])*m1
-            @inbounds for i = 2:length(s_fft)-1
-                v = abs2(s_fft[i])*m1
-                out[offset+i] += v
-                out[offset+nfft-i+2] += v
+            out[offset+1] = muladd(abs2(s_fft[1]), m1, out[offset+1])
+            @inbounds for i = 2:n-1
+                k = abs2(s_fft[i])
+                out[offset+i] = muladd(k, m1, out[offset+i])
+                out[offset+nfft-i+2] = muladd(k, m1, out[offset+nfft-i+2])
             end
-            out[offset+n] += abs2(s_fft[n])*m1
+            out[offset+n] = muladd(abs2(s_fft[n]), m1, out[offset+n])
             if isodd(nfft)
-                out[offset+n+1] += abs2(s_fft[n])*m1
+                out[offset+n+1] = muladd(abs2(s_fft[n]), m1, out[offset+n+1])
             end
         end
     end
     out
 end
 
 # Convert the output of a 2-d FFT to a 2-d PSD and add it to out
-function fft2pow2!(out::Matrix{T}, s_fft::Matrix{Complex{T}}, n1::Int, n2::Int, r::Real) where T
+function fft2pow2!(out::Matrix{T}, s_fft::Matrix{Complex{T}}, r::Real) where T
     m1 = convert(T, 1/r)
-    for j = 1:n2
-        for i = 1:n1
-            @inbounds out[i,j] += abs2(s_fft[i,j])*m1
-        end
+    for i in eachindex(out, s_fft)
+        @inbounds out[i] = abs2(s_fft[i]) * m1
```

Comment:
In this `@inbounds` helpful? I'd imagine the compiler can eliminate or at lest hoist it out of the loop.